mev-relay-rs/src/relay.rs

```diff
@@ -618,19 +602,18 @@ impl BlindedBlockRelayer for Relay {
 
             self.validate_builder_submission_trusted(bid_trace, signed_submission.payload())?;
             debug!(%auction_request, "validated builder submission");
-            (auction_request, bid_trace.value, bid_trace.builder_public_key.clone())
+            (auction_request, bid_trace.value)
         };
 
         let message = signed_submission.message();
         let public_key = &signed_submission.message().builder_public_key;
         let signature = signed_submission.signature();
         verify_signed_builder_data(message, public_key, signature, &self.context)?;
 
-        let execution_payload = signed_submission.payload().clone();
         // NOTE: this does _not_ respect cancellations
         // TODO: move to regime where we track best bid by builder
         // and also move logic to cursor best bid for auction off this API
-        self.insert_bid_if_greater(auction_request, execution_payload, value, builder_public_key)?;
+        self.insert_bid_if_greater(auction_request, signed_submission, value)?;
```

Comment:
yeah, nicer to just pass in the `signed_submission` here then let this fn decide what to do